addons/stock_account/models/product.py

```diff
@@ -164,6 +164,13 @@ def _compute_value(self):
                 product.total_value = product._run_avco(at_date=at_date)[1] * qty_available / total_qty_available
             else:
                 product.total_value = product.with_context(warehouse_id=False)._run_fifo(total_qty_available, at_date=at_date) * qty_available / total_qty_available
+
+    @api.depends_context('to_date', 'company', 'warehouse_id')
+    @api.depends("total_value")
+    def _compute_avg_cost(self):
+        for product in self:
+            valuated_product = product.sudo(False)._with_valuation_context()
+            qty_available = valuated_product.qty_available
             if product.company_currency_id.is_zero(product.total_value):
                 product.avg_cost = product._get_standard_price_at_date()
             else:
```

Comment:
```suggestion
            valuated_product = product.sudo(False)._with_valuation_context()
            if product.company_currency_id.is_zero(product.total_value):
                product.avg_cost = product._get_standard_price_at_date()
            else:
                qty_available = valuated_product.qty_available
```

`qty_available` is computed. Try looking up its field definition+ its compute method.